src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -204,111 +238,405 @@ describe('lambda flare', () => {
+
+  describe('createDirectories', () => {
+    const MOCK_LOG_PATH = path.join(MOCK_FOLDER_PATH, 'logs')
+    it('successfully creates a root folder', async () => {
+      createDirectories(MOCK_FOLDER_PATH, MOCK_LOG_PATH, new Map())
+
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+    })
+
+    it('successfully creates a root and logs folder', async () => {
+      createDirectories(MOCK_FOLDER_PATH, MOCK_LOG_PATH, MOCK_LOGS)
 
-      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
       expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
-      expect(fs.writeFileSync).toHaveBeenCalledWith(MOCK_FILE_PATH, MOCK_DATA)
+      expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_LOG_PATH)
     })
 
-    it('throws error when unable to create folder', async () => {
+    it('throws error when unable to create a folder', async () => {
       ;(fs.mkdirSync as jest.Mock).mockImplementation(() => {
         throw new Error('MOCK ERROR: Unable to create folder')
       })
 
-      await expect(writeFile(MOCK_FOLDER_PATH, MOCK_FILE_PATH, MOCK_DATA, context)).rejects.toMatchSnapshot()
-
-      expect(fs.existsSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
+      expect(() => createDirectories(MOCK_FOLDER_PATH, MOCK_LOG_PATH, new Map())).toThrowErrorMatchingSnapshot()
       expect(fs.mkdirSync).toHaveBeenCalledWith(MOCK_FOLDER_PATH)
-      ;(fs.mkdirSync as jest.Mock).mockRestore()
+      fs.mkdirSync = jest.fn().mockImplementation(() => {})
+    })
+  })
+
+  describe('getLogStreamNames', () => {
+    beforeEach(() => {
+      CLOUDWATCH_CLIENT_MOCK.reset()
+      mockCloudWatchLogsClientCommands(CLOUDWATCH_CLIENT_MOCK)
+    })
+
+    it('returns the 3 latest log stream names sorted by last event time', async () => {
+      const mockStreams: LogStream[] = [
+        {logStreamName: 'Stream3'},
+        {logStreamName: 'Stream2'},
+        {logStreamName: 'Stream1'},
+      ]
+      mockCloudWatchStreams(CLOUDWATCH_CLIENT_MOCK, mockStreams)
+
+      const expectedLogStreams = ['Stream1', 'Stream2', 'Stream3']
+      const logStreams = await getLogStreamNames(new CloudWatchLogsClient({}), MOCK_LOG_GROUP)
+
+      expect(logStreams).toEqual(expectedLogStreams)
+    })
+
+    it('returns undefined when no log streams are found', async () => {
+      mockCloudWatchStreams(CLOUDWATCH_CLIENT_MOCK, [])
+
+      const logStreams = await getLogStreamNames(new CloudWatchLogsClient({}), MOCK_LOG_GROUP)
+
+      expect(logStreams).toEqual([])
+    })
+
+    it('throws error when log streams cannot be retrieved', async () => {
+      mockCloudWatchStreams(CLOUDWATCH_CLIENT_MOCK, [])
+      const mockCwlClient = {
+        send: jest.fn().mockRejectedValue(new Error('Cannot retrieve log streams')),
+      }
+
+      await expect(
+        getLogStreamNames((mockCwlClient as unknown) as CloudWatchLogsClient, MOCK_LOG_GROUP)
+      ).rejects.toThrow('Cannot retrieve log streams')
+      expect(mockCwlClient.send).toHaveBeenCalled()
```

Comment:
This is not the way of mocking an error for an AWS mock client.

Reply:
Good catch, changed to `cloudWatchLogsClientMock.on(DescribeLogStreamsCommand).rejects('Cannot retrieve log streams')`